File: problem_030.swift
#!/usr/bin/xcrun swift
// -*- compile-command: "swiftc problem_030.swift -o problem_030" -*-
// Copyright (c) 2016 Michael Caldwell
import Cocoa

print("Project Euler - Problem 30:")
print("Find the sum of all the numbers that can be written as the sum of fifth powers of their digits\n")

func powerDigitSum(var n:Int) -> Int {
    var sum = 0
    while n > 0 {
        let d = n % 10
        n /= 10
        sum += Int(pow(Double(d), Double(5)))
    }
    return sum
}

var sum = 0

for i in 10...355000 {
    if i == powerDigitSum(i) {
        sum += i
    }
}

print(sum)

File: problem_030.rs
// -*- compile-command: "rustc -o problem_030_rs problem_030.rs" -*-
// Copyright (c) 2016 Michael Caldwell

fn power_digits(n: usize) -> usize {
    let mut n = n;
    let mut sum = 0;
    while n > 0 {
        let d = n % 10;
        sum += (d as f64).powi(5) as usize;
        n /= 10;
    }
    return sum;
}

fn main() {
    println!("Project Euler - Problem 30:");
    println!("Find the sum of all the numbers that can be written as the sum of fifth powers of their digits\n");

    let mut sum = 0;

    for n in 2..1000000 {
        if n == power_digits(n) {
            sum += n;
        }
    }

    println!("{}", sum);
}
